pkg/datapath/loader/cache.go

```diff
@@ -72,6 +109,111 @@ func (o *ObjectCache) Update(nodeCfg *datapath.LocalNodeConfiguration) {
 	o.baseHash = newHash
 }
 
+// serialize finds the channel that serializes builds against the same hash.
+// Returns the channel and whether or not the caller needs to compile the
+// datapath for this hash.
+func (o *ObjectCache) serialize(hash string) (fq *serializer.FunctionQueue, found bool) {
+	o.Lock()
+	defer o.Unlock()
+
+	fq, compiled := o.compileQueue[hash]
+	if !compiled {
+		fq = serializer.NewFunctionQueue(1)
+		o.compileQueue[hash] = fq
+	}
+	return fq, compiled
+}
+
+func (o *ObjectCache) lookup(hash string) (string, bool) {
+	o.Lock()
+	defer o.Unlock()
+	path, exists := o.toPath[hash]
+	return path, exists
+}
+
+func (o *ObjectCache) insert(hash, objectPath string) {
+	o.Lock()
+	defer o.Unlock()
```

Comment:
same here

Reply:
Now I'm curious if how munch you gain without the defer here. Did you write any benchmarks for this so we can compare it?

Reply:
There's the `BenchmarkCompileOrLoad()`.

For what it's worth, I changed all the defer stuff and it wasn't clear to me that there was any difference. Subsequent runs on my local laptop range from about 6-8ms/op no matter which approach I took.

Reply:
One more note on insert here, this is rare - this requires completely different datapath configuration. If this is hit often, then the entire feature is broken.